plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/LibraryClasspathContainerResolverService.java

```diff
@@ -113,16 +113,18 @@ public IStatus resolveContainer(IJavaProject javaProject, IPath containerPath,
     
     try {
       String libraryId = containerPath.segment(1);
-      Library library = CloudLibraries.getLibrary(libraryId);
+      Library library = null;
+      if (CloudLibraries.MASTER_CONTAINER_ID.equals(libraryId)) {
```

Comment:
I believe this code should be something like:
```
Library library = CloudLibraries.getLibrary(javaProject, libraryId);
```
`getLibrary()` will check and return a library definition for `libraryId`, and if not will attempt to load a serialized library definition from a file generated from that `libraryId`.

Reply:
Exactly.